### PR TITLE
style: 💄 Update Refresh Button to animate when loading

### PR DESCRIPTION
**Pull request by cameronperera**
## Description
Add animation styles to refresh button while loading

:desktop_computer: [Desktop preview](https://boundary-ui-desktop-git-refresh-button-spinner-styles-hashicorp.vercel.app/scopes/global/projects/targets)

### Screenshots (if appropriate):
correct-loading-spin

### CODE REVIEW

File: addons/rose/addon/styles/rose/components/_icon.scss [hunk 1]
```diff
@@ -17,3 +17,19 @@
     vertical-align: text-top;
   }
 }
+
+.is-loading .rose-icon {
```
Comment (randallmorey) [inline addons/rose/addon/styles/rose/components/_icon.scss:21]: I'm thinking this is not well-scoped enough to live here.  Can we move this into the applications for now?  It's sort of temporary until we fully adopt HDS.

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: addons/rose/addon/styles/rose/components/_icon.scss (hunk 1 above)
Comment (cameronperera) [reply addons/rose/addon/styles/rose/components/_icon.scss:21]: Absolutely. I will move it to applications.

Comment (cameronperera) [reply addons/rose/addon/styles/rose/components/_icon.scss]: Fixed.

**Review by ZedLi — APPROVED** [review]
LGTM!

**Review by randallmorey — APPROVED** [review]

**Pull request merged**